File: src/infrastructure/entities/Product_Image.entity.ts
import { Entity, PrimaryGeneratedColumn, Column, ManyToOne, JoinColumn } from "typeorm";

import ProductEntity from "./Product.entity";
import ProductOptionEntity from "./Product_Option.entity";

@Entity("product_images")
export default class ProductImageEntity {
  @PrimaryGeneratedColumn("increment")
  id: number;

  @ManyToOne(() => ProductEntity, { onDelete: "CASCADE" })
  @JoinColumn({ name: "product_id" })
  product: ProductEntity;

  @Column({ type: "varchar", length: 255 })
  url: string;

  @Column({ type: "varchar", length: 255, nullable: true })
  alt_text: string;

  @Column({ type: "boolean", default: false })
  is_primary: boolean;

  @Column({ type: "int", default: 0 })
  display_order: number;

  @ManyToOne(() => ProductOptionEntity, {
    onDelete: "SET NULL",
    nullable: true,
  })
  @JoinColumn({ name: "option_id" })
  option: ProductOptionEntity;
}

File: src/main/java/com/example/demo/product/entity/ProductImageEntity.java
package com.example.demo.product.entity;

import com.example.demo.product.controller.request.AddProductImageRequest;
import com.example.demo.product.controller.request.ProductImageRequest;
import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@Table(name = "product_images")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor(access = AccessLevel.PRIVATE)
public class ProductImageEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @ManyToOne(fetch = FetchType.LAZY, cascade = CascadeType.REMOVE)
    @JoinColumn(name = "product_id")
    private ProductEntity productEntity;
    @Column(name = "url", columnDefinition = "VARCHAR(255)", nullable = false)
    private String url;
    @Column(name = "alt_text", columnDefinition = "VARCHAR(255)")
    private String altText;
    @Column(name = "is_primary", columnDefinition = "BOOLEAN DEFAULT FALSE")
    private Boolean isPrimary;
    @Column(name = "display_order", columnDefinition = "INTEGER DEFAULT 0")
    private Integer displayOrder;
    @Column(name = "option_id", columnDefinition = "BIGINT")
    private Long optionId;

    public static ProductImageEntity create(ProductImageRequest request, ProductEntity productEntity) {
        return new ProductImageEntity(
                null,
                productEntity,
                request.url(),
                request.altText(),
                request.isPrimary(),
                request.displayOrder(),
                request.optionId()
        );
    }

    public static ProductImageEntity create(AddProductImageRequest request, ProductEntity productEntity) {
        return new ProductImageEntity(
                null,
                productEntity,
                request.url(),
                request.altText(),
                request.isPrimary(),
                request.displayOrder(),
                request.optionId()
        );
    }
}
